modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-renderer/src/main/resources/META-INF/resources/js/components/PageRenderer/Actions.es.js

```diff
@@ -0,0 +1,187 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+import {ClayButtonWithIcon} from '@clayui/button';
+import {ClayDropDownWithItems} from '@clayui/drop-down';
+import classNames from 'classnames';
+import domAlign from 'dom-align';
+import React, {
+	createContext,
+	forwardRef,
+	useContext,
+	useEffect,
+	useLayoutEffect,
+	useState,
+} from 'react';
+
+import {EVENT_TYPES} from '../../actions/eventTypes.es';
+import {useForm} from '../../hooks/useForm.es';
+import {useResizeObserver} from '../../hooks/useResizeObserver.es';
+
+const ActionsContext = createContext({});
+
+/**
+ * ActionsContext is responsible for store which field is being hovered or active
+ */
+export const ActionsProvider = ({children}) => {
+	const [hoveredField, setHoveredField] = useState('');
+	const [activeField, setActiveField] = useState('');
+	const dispatch = useForm();
+
+	useEffect(() => {
+		if (hoveredField) {
+			return dispatch({
+				payload: {fieldName: hoveredField},
+				type: EVENT_TYPES.FIELD_HOVERED,
+			});
+		}
+	}, [hoveredField, dispatch]);
+
+	return (
+		<ActionsContext.Provider
+			value={{activeField, hoveredField, setActiveField, setHoveredField}}
+		>
+			{children}
+		</ActionsContext.Provider>
+	);
+};
+
+export const useActions = () => {
+	return useContext(ActionsContext);
+};
+
+ActionsContext.displayName = 'ActionsContext';
+
+const ACTIONS_CONTAINER_OFFSET = [-2, 1];
+
+export const ActionsControls = forwardRef(
+	({activePage, children, columnRef, editable, field}, actionsRef) => {
```

Comment:
Since we split it in two, the `actionsRef` should not be passed as a component ref, but we should receive it as an exclusive props for this case, the prop can carry the same name. In that case we can give the feeling that we will have the reference of this component which is not true, we want the reference of another component.

Reply:
I identified the issue here. I will not forward ActionsControls ref more and just receives the `actionsRef` value from the props.